### PR TITLE
feat: change budget alerts to alarm by forecast

### PR DESCRIPTION
**Pull request by romanini-ciandt**
The goal of this PR is to change the behavior of all the existing budget alerts.

Modified threshold rules: a single alarm when 120% of forecasted is spent;
Current threshold rules: multiple alarms (50%, 75%, 90%, 95%) based on the actual value spent.

### CODE REVIEW

File: 1-org/envs/shared/variables.tf [hunk 1]
```diff
@@ -115,32 +115,41 @@ variable "project_budget" {
   budget_amount: The amount to use as the budget.
   alert_spent_percents: A list of percentages of the budget to alert on when threshold is exceeded.
   alert_pubsub_topic: The name of the Cloud Pub/Sub topic where budget related messages will be published, in the form of `projects/{project_id}/topics/{topic_id}`.
+  alert_spend_basis: The type of basis used to determine if spend has passed the threshold.
```
Comment (rjerrems) [inline 1-org/envs/shared/variables.tf:118]: Can you mention that the default is `FORCASTED_SPEND` and what the other possible values are?

File: 2-environments/modules/env_baseline/variables.tf [hunk 2]
```diff
@@ -40,20 +40,25 @@ variable "project_budget" {
   budget_amount: The amount to use as the budget.
   alert_spent_percents: A list of percentages of the budget to alert on when threshold is exceeded.
   alert_pubsub_topic: The name of the Cloud Pub/Sub topic where budget related messages will be published, in the form of `projects/{project_id}/topics/{topic_id}`.
+  alert_spend_basis: The type of basis used to determine if spend has passed the threshold.
```
Comment (rjerrems) [inline 2-environments/modules/env_baseline/variables.tf:43]: Same as above

File: 4-projects/business_unit_1/shared/variables.tf [hunk 3]
```diff
@@ -26,11 +26,13 @@ variable "project_budget" {
   budget_amount: The amount to use as the budget.
   alert_spent_percents: A list of percentages of the budget to alert on when threshold is exceeded.
   alert_pubsub_topic: The name of the Cloud Pub/Sub topic where budget related messages will be published, in the form of `projects/{project_id}/topics/{topic_id}`.
+  alert_spend_basis: The type of basis used to determine if spend has passed the threshold.
```
Comment (rjerrems) [inline 4-projects/business_unit_1/shared/variables.tf:29]: Same as above

File: 4-projects/business_unit_2/shared/variables.tf [hunk 4]
```diff
@@ -26,11 +26,13 @@ variable "project_budget" {
   budget_amount: The amount to use as the budget.
   alert_spent_percents: A list of percentages of the budget to alert on when threshold is exceeded.
   alert_pubsub_topic: The name of the Cloud Pub/Sub topic where budget related messages will be published, in the form of `projects/{project_id}/topics/{topic_id}`.
+  alert_spend_basis: The type of basis used to determine if spend has passed the threshold.
```
Comment (rjerrems) [inline 4-projects/business_unit_2/shared/variables.tf:29]: Same as above.

File: 4-projects/modules/single_project/variables.tf [hunk 5]
```diff
@@ -114,11 +114,13 @@ variable "project_budget" {
   budget_amount: The amount to use as the budget.
   alert_spent_percents: A list of percentages of the budget to alert on when threshold is exceeded.
   alert_pubsub_topic: The name of the Cloud Pub/Sub topic where budget related messages will be published, in the form of `projects/{project_id}/topics/{topic_id}`.
+  alert_spend_basis: The type of basis used to determine if spend has passed the threshold.
```
Comment (rjerrems) [inline 4-projects/modules/single_project/variables.tf:117]: Same as above.

File: 1-org/envs/shared/variables.tf (hunk 1 above)
Comment (romanini-ciandt) [reply 1-org/envs/shared/variables.tf]: Done! Let me know if the comment matches the expectation

Comment (rjerrems) [reply 1-org/envs/shared/variables.tf]: LGTM

**Pull request merged**